baselines/hfedxgboost/README.md

```diff
@@ -0,0 +1,278 @@
+
+# If you are using a dataset that requires a complicated download (i.e. not using one natively supported by TF/PyTorch) + preprocessing logic, you might want to tell people to run one script first that will do all that. Please ensure the download + preprocessing can be configured to suit (at least!) a different download directory (and use as default the current directory). The expected command to run to do this is:
+
+poetry run -m <baseline-name>.dataset_preparation <optional arguments, but default should always run>
+
+# It is expected that you baseline supports more than one dataset and different FL settings (e.g. different number of clients, dataset partitioning methods, etc). Please provide a list of commands showing how these experiments are run. Include also a short explanation of what each one does. Here it is expected you'll be using the Hydra syntax to override the default config.
+
+poetry run -m <baseline-name>.main  <override_some_hyperparameters>
+.
+.
+.
+poetry run -m <baseline-name>.main  <override_some_hyperparameters>
+```
+
+
+## Expected Results
+
+
+```bash
+#results in table 2
+python -m hfedxgboost.main --multirun clients="a9a_2_clients","a9a_5_clients","a9a_10_clients" dataset=a9a
+
+#results in table 3
+python -m hfedxgboost.main --multirun clients="cod_rna_2_clients","cod_rna_5_clients","cod_rna_10_clients" dataset=cod_rna
+
+#results in table 4
+python -m hfedxgboost.main --multirun clients="ijcnn1_2_clients","ijcnn1_5_clients","ijcnn1_10_clients" dataset=ijcnn1
+
+#results in table 5
+python -m hfedxgboost.main --multirun clients="space_ga_2_clients","space_ga_5_clients","space_ga_10_clients" dataset=space_ga
+
+#results in table 6
+python -m hfedxgboost.main --multirun clients="abalone_2_clients","abalone_5_clients","abalone_10_clients" dataset=abalone
+
+#results in table 7
+python -m hfedxgboost.main --multirun clients="cpusmall_2_clients","cpusmall_5_clients","cpusmall_10_clients" dataset=cpusmall
+
+```
+### Table 1
+
+| Dataset | task type | test result | 
+| :---: | :---: | :---: |
+| a9a | Binary classification | .84 |
+| cod-rna | Binary classification | .97 |
+| ijcnn1 | Binary classification | .98 |
+| abalone | Regression | 4.6 |
+| cpusmall | Regression | 9 |
+| space_ga | Regression | .032 |
+| YearPredictionMSD | Regression | 76.41 |
+
+### Those results don't come from following the original paper hyper-parameters, the new hyper-parameters are specified in the config files in the `clients` folder
+
+### Table 2 a9a dataset
+|  no.of clients | server-side test Accuracy 
+| :---: | :---: |
+| 2 | .84
+| 5 | .84
+| 10 | .83
+### Table 3 cod_rna dataset
+|  no.of clients | server-side test Accuracy 
+| :---: | :---: |
+| 2 | .96
+| 5 | .96
+| 10 | .95
+### Table 4 ijcnn1 dataset
+|  no.of clients | server-side test Accuracy 
+| :---: | :---: |
+| 2 | .98
+| 5 | .97
+| 10 | .96
```

Comment:
```suggestion
### Binary Classification Datasets

| Dataset |  no. of clients | server-side test Accuracy |
| :---: | :---: | :---: |
| a9a | 2<br>5<br>10 | 0.84<br>0.84<br>0.83 |
| cod_rna | 2<br>5<br>10 | 0.96<br>0.96<br>0.95 | 
| ijcnn1 | 2<br>5<br>10 | 0.98<br>0.97<br>0.96 |

```

Reply:
on the same line as my other comment earlier, maybe these three tables could be merged into a single one? Maybe there is a way to also include the regression datasets (but now i can't think of a clean way of doing it). 

But go ahead with these changes if you want and depending on what other results are you planning to add (if any).